### PR TITLE
Minimize a subset of the containers

### PR DESCRIPTION
**Pull request by rustylynch**
This pull request uses the docker multistage build technique to create a new container called clearlinux/os-core that only contains os-core, and then converts redis, httpd, cgi, nginx, and memcached to use os-core as the base layer while also using a multistage build to pull in additional content (since the base layer does not have os-core-update installed.)

### CODE REVIEW

File: redis/Dockerfile [hunk 1]
```diff
@@ -1,12 +1,29 @@
...
+FROM clearlinux:latest AS builder
+
+# Move to latest Clear Linux release to ensure
+# that the swupd command line arguments are
+# correct
+RUN swupd update --no-boot-update $swupd_args
+
+# Grab os-release info from the minimal base image so
+# that the new content matches the exact OS version
+COPY --from=clearlinux/os-core:latest /usr/lib/os-release /
```
Comment (bryteise) [inline redis/Dockerfile]: I think we could get into a weird position where the swupd os-install below fails if the clearlinux/os-core:latest is not yet updated when this unit is updated. It is likely safer to source the os-release from os-core first and run update with -V ${VERSION_ID}. I will say this problem is more theoretical for our Dockerfiles but if other folks reuse the pattern they could more readily run into trouble.

Comment (rustylynch) [reply redis/Dockerfile]: So you mean something on the lines of...
COPY --from=clearlinux/os-core:latest /usr/lib/os-release /
RUN source /os-release && swupd update --no-boot-update $swupd_args

Comment (bryteise) [reply redis/Dockerfile]: Yes ```RUN source /os-release && swupd update --no-boot-update -V ${VERSION_ID} $swupd_args```

**Comment by bryteise** [comment]
The CI is failing due to the os-core image not yet being available. It may be worth splitting this PR up but I think this is low risk enough to go without the CI.

**Comment by bryteise** [comment]
Change looks good to me overall.

**Comment by rustylynch** [comment]
I'll split this into two pulls with the minor change noted above

**Pull request closed without merging**